### PR TITLE
code refactor, added generic functions

### PR DESCRIPTION
**Pull request by redhatHameed**
Signed-off-by: Abdul Hameed <ahameed@redhat.com>

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist (Definition of Done)

- [ ] All acceptance criteria specified in JIRA or in issue number have been completed
- [ ] Unit tests added that prove the fix is effective or the feature works 
- [ ] Documentation added for the feature
- [ ] all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer

### CODE REVIEW

File: controllers/reconcilers/olm_helper.go [hunk 1]
```diff
@@ -54,3 +56,46 @@ func GetDBaaSOperatorCSV(namespace string, name string, ctx context.Context, ser
...
+func GetCatalogSource(namespace string, name string) *v1alpha1.CatalogSource {
+
+	return &v1alpha1.CatalogSource{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+	}
+
+}
```
Comment (tchughesiv) [inline controllers/reconcilers/olm_helper.go:101]: please add some simple unit-tests for these generics... i've noticed a few of your method calls for providers weren't correct.

**Review by tchughesiv — CHANGES_REQUESTED** [review]
lgtm

File: controllers/reconcilers/mongodb_atlas_installation/mongodb_atlas_installation_reconciler.go [hunk 2]
```diff
@@ -68,13 +67,13 @@ func (r *Reconciler) Reconcile(ctx context.Context, cr *v1.DBaaSPlatform, status
 
 func (r *Reconciler) Cleanup(ctx context.Context, cr *v1.DBaaSPlatform) (v1.PlatformsInstlnStatus, error) {
 
-	subscription := r.getMongoDBAtlasSubscription(cr)
+	subscription := reconcilers.GetSubscription("mongodb-atlas-subscription", cr.Namespace)
```
Comment (tchughesiv) [inline controllers/reconcilers/mongodb_atlas_installation/mongodb_atlas_installation_reconciler.go:70]: these inputs should be reversed

File: controllers/reconcilers/mongodb_atlas_installation/mongodb_atlas_installation_reconciler.go [hunk 3]
```diff
@@ -68,13 +67,13 @@ func (r *Reconciler) Reconcile(ctx context.Context, cr *v1.DBaaSPlatform, status
...
 	err := r.client.Delete(ctx, subscription)
 	if err != nil && !errors.IsNotFound(err) {
 		return v1.ResultFailed, err
 	}
 
-	catalogSource := r.getMongoDBAtlasCatalogSource()
+	catalogSource := reconcilers.GetCatalogSource("mongodb-atlas-catalogsource", reconcilers.CATALOG_NAMESPACE)
```
Comment (tchughesiv) [inline controllers/reconcilers/mongodb_atlas_installation/mongodb_atlas_installation_reconciler.go:76]: these inputs should be reversed

File: controllers/reconcilers/mongodb_atlas_installation/mongodb_atlas_installation_reconciler.go [hunk 4]
```diff
@@ -107,8 +106,8 @@ func (r *Reconciler) Cleanup(ctx context.Context, cr *v1.DBaaSPlatform) (v1.Plat
 }
 func (r *Reconciler) reconcileSubscription(cr *v1.DBaaSPlatform, ctx context.Context) (v1.PlatformsInstlnStatus, error) {
 
-	subscription := r.getMongoDBAtlasSubscription(cr)
-	catalogsource := r.getMongoDBAtlasCatalogSource()
+	subscription := reconcilers.GetSubscription(reconcilers.MONGODB_ATLAS_CSV, cr.Namespace)
+	catalogsource := reconcilers.GetCatalogSource("mongodb-atlas-catalogsource", reconcilers.CATALOG_NAMESPACE)
```
Comment (tchughesiv) [inline controllers/reconcilers/mongodb_atlas_installation/mongodb_atlas_installation_reconciler.go:110]: these inputs should be reversed

**Review by tchughesiv — CHANGES_REQUESTED** [review]

File: controllers/reconcilers/crunchybridge_installation/crunchybridge_installation_reconciler.go [hunk 5]
```diff
@@ -107,8 +106,8 @@ func (r *Reconciler) Cleanup(ctx context.Context, cr *v1.DBaaSPlatform) (v1.Plat
 
 func (r *Reconciler) reconcileSubscription(cr *v1.DBaaSPlatform, ctx context.Context) (v1.PlatformsInstlnStatus, error) {
 
-	subscription := r.getCrunchyBridgeSubscription(cr)
-	catalogsource := r.getCrunchyBridgeCatalogSource()
+	subscription := reconcilers.GetSubscription(cr.Namespace, "crunchy-bridge-subscription")
+	catalogsource := reconcilers.GetCatalogSource(reconcilers.CATALOG_NAMESPACE, "crunchy-bridge-catalogsource")
```
Comment (tchughesiv) [inline controllers/reconcilers/crunchybridge_installation/crunchybridge_installation_reconciler.go:110]: given that `"crunchy-bridge-catalogsource"` is used more than once, it should probably be a global var, at least for the reconcilers package. thoughts? i'd imagine there are several other strings we should consider as well?

**Review by tchughesiv — CHANGES_REQUESTED** [review]

File: controllers/reconcilers/olm_helper.go (hunk 1 above)
Comment (redhatHameed) [reply controllers/reconcilers/olm_helper.go:101]: added

File: controllers/reconcilers/mongodb_atlas_installation/mongodb_atlas_installation_reconciler.go (hunk 3 above)
Comment (redhatHameed) [reply controllers/reconcilers/mongodb_atlas_installation/mongodb_atlas_installation_reconciler.go]: updated

File: controllers/reconcilers/crunchybridge_installation/crunchybridge_installation_reconciler.go (hunk 5 above)
Comment (redhatHameed) [reply controllers/reconcilers/crunchybridge_installation/crunchybridge_installation_reconciler.go:110]: Agree. Let me update them

Comment (redhatHameed) [reply controllers/reconcilers/crunchybridge_installation/crunchybridge_installation_reconciler.go]: updated

**Review by tchughesiv — APPROVED** [review]
lgtm

**Pull request merged**